### PR TITLE
Updated xp rates for mahogany homes including any combination of outfit pieces

### LINKED ISSUE #53: Not updating construction xp with carpenter's outfit modifier when then activity is mahogany homes
As title says.

When selecting the carpenters outfit modifier, it only updates the 'total banked xp' when the activity 'products' is selected. When the activity 'mahogany homes' is selected, the carpenters outfit does not affect the banked xp.

### PR DESCRIPTION
**Pull request by SeaifanAladdin**
As the mechanics of mahogany homes was recently been discovered in more details, the average xp, planks and steelbars per contract has updated on the wiki
https://oldschool.runescape.wiki/w/Mahogany_Homes
https://oldschool.runescape.wiki/w/Mahogany_Homes/Mechanics

I have built on the Wiki's work by determining a formula to calculate the xp rate per plank for each contract, accounting for any combination of the outfit pieces. 

The formula and parameterized values used in CarpentersOutfit.java was derived from 
https://docs.google.com/spreadsheets/d/1573_E5suKS7OY2E3utGpBkuaReiF86hTh0p8PLqi2II/edit?usp=sharing

Note that to reduce duplicate code, I ended up refactoring SkillingOutfit.java's appliedXpRate method. 

Closes #53

### CODE REVIEW

**Comment by TheStonedTurtle** [comment]
I'd rather not add logging for incorrect xp rates since people who run the client aren't going to be checking their logs or even notice they are incorrect (6b49412ecfdff2ac4ed5754d06d68d583e1ec726). I'd rather not add this if we aren't sure the values are correct, but since it's already marked as an RNG activity I guess we can do it.

You have some checkstyle issues that I'm fixing and I'll remove this myself

**Comment by SeaifanAladdin** [comment]
That is fair, I did add the logging on a whim. 

Although I added the logging of the xp rates for developers rather than users. Before I had it that the base exp rate as defined in the Activity class was dynamically calculated using the exp rates in the carpenters outfit class, but realized that I did not like this type of coupling. I added the logging as some sort of compensation but I realize it's not necessary at all.

**Review by TheStonedTurtle — APPROVED** [review]

**Comment by TheStonedTurtle** [comment]
@SeaifanAladdin Please see the comment I added in 69a8478fb8cefa37db0be76ed190555798a1f679, if that makes sense to you I think we're good here.

File: src/main/java/thestonedturtle/bankedexperience/data/modifiers/CarpentersOutfit.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+				.put(Activity.MAHOGANY_HOMES_TEAK, new XpPerTask(87.735f, 200.117f))
+				.put(Activity.MAHOGANY_HOMES_MAHOGANY, new XpPerTask(127.259f, 218.861f))
+				.build();
+	}
+
+	@Override
+	public double appliedXpRate(Activity activity)
+	{
+		XpPerTask xpPerTask = xpPerMahoganyHomesActivityMap.get(activity);
+		if (xpPerTask != null)
```
[the remark this answers is not in the document]
Comment (SeaifanAladdin) [reply src/main/java/thestonedturtle/bankedexperience/data/modifiers/CarpentersOutfit.java]: I completely agree with the suggestion and change

**Pull request merged**